packages/nectar/src/sagas/errors/handleErrors/handleErrors.saga.ts

```diff
@@ -23,25 +25,31 @@ function* retryRegistration(communityId: string) {
   log(`registering certificate for community ${communityId} failed, trying again`)
 }
 
-let registrationAttempts: number = 0
-
 export function* handleErrorsSaga(
   action: PayloadAction<ReturnType<typeof errorsActions.addError>['payload']>
 ): Generator {
   const error: ErrorPayload = action.payload
+  const registrationAttempts = yield* select(
+    communitiesSelectors.registrationAttempts(error.community)
+  )
+
   if (error.type === SocketActionTypes.REGISTRAR) {
     if (error.code === ErrorCodes.FORBIDDEN) {
       yield* put(errorsActions.addError(error))
     }
-    if (error.code === ErrorCodes.NOT_FOUND || error.code === ErrorCodes.SERVER_ERROR || error.code === ErrorCodes.SERVICE_UNAVAILABLE) {
-      // 7 retries guarantees 99.9% chance of registering username if registrar is online.
-      if (registrationAttempts < 7) {
+    if (
+      error.code === ErrorCodes.NOT_FOUND ||
+      error.code === ErrorCodes.SERVER_ERROR ||
+      error.code === ErrorCodes.SERVICE_UNAVAILABLE
+    ) {
+      if (registrationAttempts < 20) {
```

Comment:
Can you explain why the magic "20" is the right number for registration attempts? :D

Reply:
It's arbitrary and gives us very high chance that registration will succeed without asking user to resubmit form

Reply:
I'll add this note in that place.